databricks/koalas/internal.py

```diff
@@ -463,22 +470,26 @@ def __init__(
         >>> internal._scol
         Column<b'(a, y)'>
         """
-        assert isinstance(sdf, spark.DataFrame)
-        assert not sdf.isStreaming, "Koalas does not support Structured Streaming."
+
+        assert isinstance(spark_frame, spark.DataFrame)
+        assert not spark_frame.isStreaming, "Koalas does not support Structured Streaming."
 
         if index_map is None:
-            assert not any(SPARK_INDEX_NAME_PATTERN.match(name) for name in sdf.columns), (
+            assert not any(SPARK_INDEX_NAME_PATTERN.match(name) for name in spark_frame.columns), (
                 "Index columns should not appear in columns of the Spark DataFrame. Avoid "
                 "index column names [%s]." % SPARK_INDEX_NAME_PATTERN
             )
 
             # Create default index.
-            sdf = _InternalFrame.attach_default_index(sdf)
-            index_map = [(SPARK_DEFAULT_INDEX_NAME, None)]
+            spark_frame = _InternalFrame.attach_default_index(spark_frame)
+            index_map = OrderedDict({SPARK_DEFAULT_INDEX_NAME: None})
 
-        if NATURAL_ORDER_COLUMN_NAME not in sdf.columns:
-            sdf = sdf.withColumn(NATURAL_ORDER_COLUMN_NAME, F.monotonically_increasing_id())
+        if NATURAL_ORDER_COLUMN_NAME not in spark_frame.columns:
+            spark_frame = spark_frame.withColumn(
+                NATURAL_ORDER_COLUMN_NAME, F.monotonically_increasing_id()
+            )
 
+        assert isinstance(index_map, OrderedDict), index_map
```

Comment:
BTW, `OrderedDict` seems not working as a type hint. So I had to use `Dict`, and add an `assert` here.